### PR TITLE
ENH: Refactor and replace plots and outputs + Compare plots

### PR DESCRIPTION
**Pull request by MateusStano**
## Pull request type

Please check the type of change your PR introduces:

- [x] Code base additions (bugfix, features)
- [x] Code maintenance (refactoring, formatting, renaming, tests)
- [ ] ReadMe, Docs and GitHub maintenance
- [ ] Other (please describe):

## What is the current behavior?  What is the new behavior?

Currently, every class has a set of methods that are used for plotting useful information. These methods add up to thousands of lines of code that could be better placed somewhere else.

The changes this PR brings can be summarized in

- create `rocketpy/plots` folder with separate files for the plot code for every class. Each of these files defines a class whose methods displays relevant plots for that class:
   -  `flight_plots.py` that holds code for `class FlightPlots()` 
   -  `environment_plots.py` that holds code for `class EnvironmentPlots()` 
   -  `rocket_plots.py` that holds code for `class RocketPlots()` 

- create `rocketpy/prints` folder with separate files for the code that prints important information for every class. Each of these files defines a class whose methods print relevant information regarding an object of the class:
   -  `flight_prints.py` that holds code for `class FlightPrints()` 
   -  `environment_prints.py` that holds code for `class EnvironmentPrints()` 
   -  `rocket_prints.py` that holds code for `class RocketPrints()` 

- Each class (`Flight`, `Environment` and `Rocket`) will have two new attributes that will objects of their respective `plot` and `print` classes.

- `rocketpy/plots/compare_plots.py` will be created. This file will contain the classes (or class) that will be responsible for comparing plots of different objects. So, if it receives two Flight objects, it will be able to plot eveything in an Flight.allInfo() but with information regarding both Flight objects. This could eventually become a way to compare simulated flights with telemetry data or with simulations from other software

## Workflow

See: https://github.com/RocketPy-Team/RocketPy/pull/287#issuecomment-1371390646

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

## Other information

It is important to note that this PR will not deal with any plots in the Motor classes since they are currently being reworked. The Motor plots will need to be refactored later.

### CODE REVIEW

**Comment by Gui-FernandesBR** [comment]
@MateusStano nice work!! Let's get our hand dirty then, I'm excited to.

My first suggestions: 
1. At `_rocketpy/__init__.py` file, I think we need to add the following code line: `from . import plots`
2. Underscoring the `FlightPlots`, `RocketPlots` and `EnvironmentPlots` classes since the user will never interact with them directly. So they would become: `_FlightPlots`, `_RocketPlots` and `_EnvironmentPlots`. Thus, inside the Flight class we should have: 

`from .plots.flight_plots import _FlightPlots`
...
    `@property`
    `def plot(self):`
    `    return _FlightPlots(self)`


This way the user would utilize:
`TestFlight.plot.trajectoy3d()`
`TestFlight.plot.acceleration()` 
...

**Comment by MateusStano** [comment]
Hey, just gonna proprose something interesting here

One way we could deal with this is actually by making plot classes (e.g. `FlightPlots`) just as we established, but instead of defining the `Flight.plot` attribute we could just make the Flight class inherit the FlightPlots class:

```
class Flight(FlightPlots):
    ...
```

This way all the plot methods will be called just as it was before, with: `TestFlight.plotSomethin()`
No breaking changes this way

**Comment by Gui-FernandesBR** [comment]
@MateusStano , I'm starting to be a little bit lost here. Could you update the task list in the PR description please?

Also, are we going to have a new meeting to follow up this one? There's other tasks that need our attention before v1, therefore finishing this one is more than important

**Comment by MateusStano** [comment]
## Workflow Update

- create a new 'flight_plots_and_prints' branch from 'new_plots' (each PR from a different branch being merge into 'flight_plots_and_prints') {Stano}
   - [x] 1st PR: moving prints from Flight.py to flight_prints.py {Stano}
   - [x] 2nd PR: moving plots from Flight.py to flight_plots.py {Stano}
 - create a new 'environment_plots_and_prints' branch from 'new_plots' {Stano}
   - [x] 1st PR: moving prints from Envronment.py to environment_prints.py {Stano}
   - [x] 2nd PR: moving plots from Envronment.py to envronment_plots.py {Stano}
- create a new 'rocket_plots_and_prints' branch from 'new_plots' {Oscar}
   - [x] 1st PR: moving prints from Rocket.py to rocket_prints.py {Oscar}
   - [x] 2nd PR: moving plots from Rocket.py to rocket_plots.py {Oscar}
- create a new 'compare_plots' branch from 'new_plots'
   - [x] 1st PR: create CompareFlight class {Gui}
- after everything is done:
    - [ ] PR: create unit tests
    - [ ] Approve/Merge into beta\v1.0.0


## To Be Done in Other PRs
   - [ ] 2nd PR: create CompareEnvironment class {Stano} --> Will be done in separate PR (maybe not for beta/v1.0)
   - [ ] 3rd PR: create CompareSimulators and/or CompareFromTelemetryData class --> Will be done in seperate PR (maybe not for beta/v1.0)
   - [ ] PR: more flexible plots and styles --> Will be done in seperate PR (maybe not for beta/v1.0)
   - [ ] PR: commit updating notebooks --> To be done when beta/v1.0 is preparing to be merged into develop




After Unit tests are done this PR will be ready to be merged

**Comment by Gui-FernandesBR** [comment]
@MateusStano , I can't understand why, but the tests stopped passing after I tried to merge the conflicts in last commit. I can check the diff [here])https://github.com/RocketPy-Team/RocketPy/commit/2b8ee0a9bdd7f3b6854a2c209bce867b0c6bb722?diff=split). I already tried to debug it but I could not find the error.

Could you help me taking a look at it please?

**Comment by Gui-FernandesBR** [comment]
> @MateusStano , I can't understand why, but the tests stopped passing after I tried to merge the conflicts in last commit. I can check the diff [here])[2b8ee0a?diff=split](https://github.com/RocketPy-Team/RocketPy/commit/2b8ee0a9bdd7f3b6854a2c209bce867b0c6bb722?diff=split)). I already tried to debug it but I could not find the error.
> 
> Could you help me taking a look at it please?

Solved! It turns out that the easiest approach was to revert the merge commit and try to merge again. We are good now.

**Review by Gui-FernandesBR — APPROVED** [review]
Tests are passing on windows and macos.
Brilliant work refactoring the plots of every main class. RocketPy is going to be improved a lot by those changes.
There are still some to-dos to be addressed in the future, but no worries for now.
And, of course, the motor classes were not touched do far.

@MateusStano go ahead and merge as soon as you few comfortable to.

File: rocketpy/Flight.py [hunk 1]
```diff
@@ -2345,9 +2374,7 @@ def retrieve_acceleration_arrays(self):
                 callback(self)
             # Loop through time steps in flight phase
             for step in self.solution:  # Can be optimized
-                if initTime < step[0] <= finalTime or (
```
Comment (Gui-FernandesBR) [inline rocketpy/Flight.py:2348]: This was (and still is, actually), the source of the problem when running the tests.
If you see something like 'cannot make function from attribute acceleration'. This code line is causing that.

**Pull request merged**